payjoin/src/send/error.rs

```diff
@@ -235,3 +236,179 @@ impl std::error::Error for CreateRequestError {
+impl Display for ResponseError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Self::WellKnown(e) => e.fmt(f),
+            // Don't display unknowns to end users, only debug logs
+            Self::Unrecognized(_, _) => write!(f, "The receiver sent an unrecognized error."),
+            Self::Validation(e) => write!(f, "The receiver sent an invalid response: {}", e),
+        }
+    }
+}
+
+impl fmt::Debug for ResponseError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Self::WellKnown(e) => write!(
+                f,
+                r#"Well known error: {{ "errorCode": "{}",
+                "message": "{}" }}"#,
+                e.error_code(),
+                e.to_string(),
+            ),
+            Self::Unrecognized(code, msg) => write!(
+                f,
+                r#"Unrecognized error: {{ "errorCode": "{}", "message": "{}" }}"#,
+                code, msg
+            ),
+            Self::Validation(e) => write!(f, "Validation({:?})", e),
+        }
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum WellKnownError {
+    Unavailable,
+    NotEnoughMoney,
+    VersionUnsupported,
+    OriginalPsbtRejected,
+}
+
+impl WellKnownError {
+    pub fn error_code(self) -> &'static str {
+        match self {
+            WellKnownError::Unavailable => "unavailable",
+            WellKnownError::NotEnoughMoney => "not-enough-money",
+            WellKnownError::VersionUnsupported => "version-unsupported",
+            WellKnownError::OriginalPsbtRejected => "original-psbt-rejected",
+        }
+    }
+}
+
+impl FromStr for WellKnownError {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<Self, ()> {
+        match s {
+            "unavailable" => Ok(WellKnownError::Unavailable),
+            "not-enough-money" => Ok(WellKnownError::NotEnoughMoney),
+            "version-unsupported" => Ok(WellKnownError::VersionUnsupported),
+            "original-psbt-rejected" => Ok(WellKnownError::OriginalPsbtRejected),
+            _ => Err(()),
+        }
+    }
+}
+
+impl Display for WellKnownError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Self::Unavailable => write!(f, "The payjoin endpoint is not available for now."),
+            Self::NotEnoughMoney => write!(f, "The receiver added some inputs but could not bump the fee of the payjoin proposal."),
+            Self::VersionUnsupported => write!(f, "This version of payjoin is not supported."),
+            Self::OriginalPsbtRejected => write!(f, "The receiver rejected the original PSBT."),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use bitcoind::bitcoincore_rpc::jsonrpc::serde_json::json;
+
+    use super::*;
+
+    #[test]
+    fn test_parse_json() {
+        let known_str_error = r#"{"errorCode":"version-unsupported", "message":"This version of payjoin is not supported."}"#;
```

Comment:
This test could make the message distinct from the template, and check that it prints the message to debug and displays the canned "This version of payjoin is not supported"